R/vars_manager.R

```diff
@@ -293,20 +297,54 @@ get_dataframe_var <- function(
 
         obj_sorted <- obj_filtered[order(attr(obj_filtered, "row.names"), decreasing=decreasing),]
         obj_pre <- `if`(abbrev, obj_sorted[1:abbrev_len,], obj_sorted)
+
     } else if (is.vector(sort_by) && length(sort_by) > 0) {
-        order_params <- obj_filtered[match(sort_by, names(obj_filtered))]
+        
+        convert_param_to_character <- function(x) {
+            # Returns "!is.na(col_name), col_name" or "desc(col_name)" in character format
+            # to be evaluated in 'dplyr::arrange()' later.
+            col_name <- x[1, "col"]
+            return(
+                `if`(
+                    x[1, "asc"],
+                    c(paste0("!is.na(", col_name, ")"), paste0(col_name)),
+                    paste0("desc(", col_name, ")")
+                )
+            )
+        }
 
+        # Parameters for arrange() function
+        params <- c()
+
+        # Create a dataframe with 'col' and 'asc' columns
+        # Each row will be a parameter in the arrange() function
         if (is.vector(ascending) && length(ascending) == length(sort_by)) {
-            i <- 0
-            for (col in order_params) {
-                i<-i+1
-                order_params[i] <- `if`(ascending[i], order_params[i], order_params[i]*-1)
-            }
-        } else if (is.logical(ascending) && ascending == FALSE) {
-            order_params <- order_params * -1
+            df_sort <- data.frame(
+                col = sort_by,
+                asc = ascending
+            )
+        } else if (is.logical(ascending) && (ascending == FALSE || ascending[0] == FALSE)) {
+            df_sort <- data.frame(
+                col = sort_by,
+                asc = FALSE
+            )
+        } else {
+            df_sort <- data.frame(
+                col = sort_by,
+                asc = TRUE
+            )
         }
 
-        obj_sorted <- obj_filtered[do.call(order, order_params),]
+        for (row in 1:nrow(df_sort)) {
+            params <- append(params, convert_param_to_character(df_sort[row,]))
+        }
```

Comment:
just in case it's not clear- I'm basically constructing a vector of which columns I want to sort. So in the end, `params` will look something like
`desc(Col1), !is.na(Col2), Col2`
which would sort by Col1 descending, and Col2 ascending. And the "na" stuff is in there because `arrange()` puts NA values at the end, regardless of whether it's ascending or descending. So when we choose ascending, I add an extra parameter in there to put NA values at the start